src/submission/tests.py

```diff
@@ -114,6 +114,63 @@ def test_article_image_galley(self):
 
         self.assertEqual(galley.file_content().strip(), expected)
 
+    def test_article_galley_unlink_files(self):
+        import pytz
+        london_timezone = pytz.timezone("Europe/London")
+        article = models.Article.objects.create(
+            journal=self.journal_one,
+            date_published=london_timezone.localize(dateparser.parse("2020-01-01")),
+            stage=models.STAGE_PUBLISHED,
+        )
+        # galley_file = File.objects.create(
+        #     article_id=article.pk,
+        #     label="image",
+        #     mime_type="image/jpeg",
+        #     is_galley=True,
+        # )
+        # galley = create_galley(article, galley_file)
+        #
+        # Using the above code, the galley_file does not have a real
+        # file on the filesystem, so the galley.unlink_files()
+        # procedure will fail because it will try to unlink a
+        # directory (the files/articles/article.id/). On the other
+        # hand, I cannot create a galley without a core.File because
+        # of DB constraints. Also, if I try to trick the system to
+        # skip the unlinking of the galley_file, making it look like
+        # it doesn't exists (which, in fact, it doesn't) with
+        # `galley.file = None` this does not work either, because it
+        # gives RelatedObjectDoesNotExist
```

Comment:
I think `os.unlink` gave me `IsADirectoryError`, but I see your point. I'll check to see if I can make the procedure to run by managing that exception.

Reply:
> I think `os.unlink` gave me `IsADirectoryError`, but I see your point. I'll check to see if I can make the procedure to run by managing that exception.

An alternative would be to populate the `file.uuid_filename` in your test, with a dummy value so that it still raises a `FileNotFoundError` as expected. The issue you are facing is probably that because there is no file name, it is constructing the relative path to the parent directory and thus cannot be unlinked like this